### PR TITLE
DOCSP-34333 mongosh v2.1.0 release notes

### PR DESCRIPTION
**Pull request by jocelyn-mendez1**
## DESCRIPTION
mongosh v2.1.0 release notes

## STAGING
https://preview-mongodbjocelynmendez1.gatsbyjs.io/mongodb-shell/DOCSP-34333/changelog/

## JIRA
https://jira.mongodb.org/browse/DOCSP-34333?filter=-1

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=6564cbd913dc9d16ea800590

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/changelog.txt [hunk 1]
```diff
@@ -12,6 +12,20 @@ Release Notes
...
+*Released November 21, 2023*
+
+- :issue:`MONGOSH-1621` & :issue:`NODE-5709` – Homebrew users who got autoupgraded 
+  to Node.js 21 stop seeing deprecation warnings. 
+
+- :issue:`MONGOSH-1452` & :issue:`NODE-5040` – ``mongosh`` now displays BSON objects in a format 
+  that is more consistent with other ``mongosh`` output. Also, many BSON objects now support syntax 
+  highlighting.
+
+- :issue:`MONGOSH-1527` – You can now iterate ``mongosh`` cursors with an idiomatic.
```
Comment (jeff-allen-mongo) [inline source/changelog.txt:27]: [Suggestion]

Suggest writing out "and", rather than using ampersand.

File: source/changelog.txt [hunk 2]
```diff
@@ -12,6 +12,20 @@ Release Notes
    :depth: 1
    :class: singlecol
 
+v2.1.0
+------
+
+*Released November 21, 2023*
+
+- :issue:`MONGOSH-1621` & :issue:`NODE-5709` – Homebrew users who got autoupgraded
```
Comment (jeff-allen-mongo) [inline source/changelog.txt:20]:
```suggestion
- :issue:`MONGOSH-1621` & :issue:`NODE-5709` – Homebrew users who were automatically upgraded 
```

File: source/changelog.txt (hunk 1 above)
Comment (jeff-allen-mongo) [inline source/changelog.txt:27]: I think what we mean here is " idiomatic syntax".

Maybe we can write this as:

You can now iterate `mongosh` cursors with idiomatic syntax: ```for (const doc of db.coll.find()) { }```.

Optionally, maybe specify that previous versions of `mongosh` only supported `.forEach` syntax for iteration.

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM, just left a couple comments. Thanks @jocelyn-mendez!

File: source/changelog.txt (hunk 1 above)
Comment (jocelyn-mendez1) [reply source/changelog.txt]: Done.

File: source/changelog.txt (hunk 2 above)
Comment (jocelyn-mendez1) [reply source/changelog.txt]: Done.

File: source/changelog.txt (hunk 1 above)
Comment (jocelyn-mendez1) [reply source/changelog.txt]: Done.

**Comment by jocelyn-mendez1** [comment]
Hi @addaleax, I have updated the mongosh release notes. Please let me know of any possible changes. Thanks in advance!

File: source/changelog.txt [hunk 3]
```diff
@@ -12,6 +12,22 @@ Release Notes
...
+- :issue:`MONGOSH-1621` and :issue:`NODE-5709` – Homebrew users who were automatically upgraded  
+  to Node.js 21 stop seeing deprecation warnings. 
+
+- :issue:`MONGOSH-1452` and :issue:`NODE-5040` – ``mongosh`` now displays BSON objects in a format 
+  that is more consistent with other ``mongosh`` output. Also, many BSON objects now support syntax 
+  highlighting.
+
+- :issue:`MONGOSH-1527` – You can now iterate mongosh cursors with idiomatic 
+  syntax: for (const doc of db.coll.find()) { }. Previously, ``mongosh`` only supported ``.forEach`` 
+  syntax for iteration.
```
Comment (addaleax) [inline source/changelog.txt:29]: The `for (const doc of db.coll.find()) { }` should probably be a code block as well?

**Review by addaleax — APPROVED** [review]
Looks good!

(No need to ping me again after changes, I’ll be OOO for a week starting in an hour :) If you need anything, feel free to ask in #mongosh)

**Pull request merged**